prov/sm2/src/sm2_progress.c

```diff
@@ -329,6 +329,21 @@ static int sm2_progress_atomic(struct sm2_ep *ep,
 		err = ret;
 	}
 out:
+
+	/* Send completion for ofi_op_atomic is generated immediately on the
+	 * sender (unless FI_DELIVERY_COMPLETE flag is set). Other ops require
+	 * delivery complete semantics, so set the SM2_GENERATE_COMPLETION flag
+	 */
+	switch (xfer_entry->hdr.op) {
+	case ofi_op_atomic:
+		if (xfer_entry->hdr.op_flags & FI_DELIVERY_COMPLETE)
+			xfer_entry->hdr.proto_flags |= SM2_GENERATE_COMPLETION;
```

Comment:
I'm a little confused about the usage of these two flags. The commit message says that one flag is basically replacing the other but you're using them together? I think the use cases for the op_flags vs proto_flags needs to be outlined a little better to make it clearer what each is used for